### PR TITLE
Docs Updates

### LINKED ISSUE #105: Docs - these pages need updating
1. https://docs.defang.io/docs/concepts/defang-byoc
2. https://docs.defang.io/docs/concepts/debug

### PR DESCRIPTION
**Pull request by raphaeltm**
Closes #105

### CODE REVIEW

File: docs/concepts/debug.md [hunk 1]
```diff
@@ -10,4 +10,16 @@ Defang includes an AI-driven tool to help you debug your cloud applications. The
...
 :::warning
 The AI debugging assistant is currently in preview and is currently limited in its capabilities. We plan to expand the capabilities of the AI assistant in the future.
+:::
+
+## How it works
+
+When you deploy a service to the [Defang Playground](./defang-playground.md), the Defang CLI will wait for the service's status to switch to healthy. If the service fails to deploy, the AI debugger will kick in. It will analyze the logs and files in your project to identify the issue and provide you with a suggested fix.
+
+:::warning
+The AI debugger currently only works in Playground.
```
Comment (Prakash-Sundaresan) [inline docs/concepts/debug.md:20]: Say something about Debug on BYOC coming soon.

File: docs/concepts/debug.md [hunk 2]
```diff
@@ -10,4 +10,16 @@ Defang includes an AI-driven tool to help you debug your cloud applications. The
...
+:::
+
+:::info
+The AI debugger only kicks in when the service fails to deploy. At the moment, we do not offer any way to trigger the AI debugger manually.
```
Comment (Prakash-Sundaresan) [inline docs/concepts/debug.md:24]: Is this true? Doesn't it kick in if the Deploy succeeds but healthcheck fails?

**Review by Prakash-Sundaresan — CHANGES_REQUESTED** [review]
See detailed comments

File: docs/concepts/debug.md (hunk 2 above)
Comment (raphaeltm) [reply docs/concepts/debug.md:24]: What you're describing is what I would call a failed deployment.

Comment (raphaeltm) [reply docs/concepts/debug.md:24]: Also, I just double checked with @lionello and it gets triggered for other deployment failures as well: "anything except compose syntax errors"

Comment (raphaeltm) [reply docs/concepts/debug.md]: @Prakash-Sundaresan I updated with more details about different potential failure cases.

**Pull request merged**